age-plugin/src/recipient.rs

```diff
@@ -327,23 +373,58 @@ pub(crate) fn run_v1<P: RecipientPluginV1>(mut plugin: P) -> io::Result<()> {
         |index, plugin_name, bytes| plugin.add_identity(index, plugin_name, &bytes),
     );
 
+    let required_labels = plugin.labels();
+
+    let labels = match (labels_supported, required_labels.is_empty()) {
+        (Ok(true), _) | (Ok(false), true) => {
+            if required_labels.contains("") {
+                Err(vec![Error::Internal {
+                    message: "Plugin tried to use the empty string as a label".into(),
+                }])
+            } else if required_labels.iter().all(is_arbitrary_string) {
+                Ok(required_labels)
+            } else {
+                Err(vec![Error::Internal {
+                    message: "Plugin tried to use a label containing an invalid character".into(),
+                }])
+            }
+        }
+        (Ok(false), false) => Err(vec![Error::Internal {
+            message: "Plugin requires labels but client does not support them".into(),
```

Comment:
Hmm, this is an interesting decision, should a postquantum plugin refuse to work with older versions of age(1)?

Reply:
That was my interpretation, given this language in https://github.com/C2SP/C2SP/pull/37:

>   - Clients MUST check that all recipient stanzas wrapping a given file key have the exact same label set. Clients MUST NOT permit partial overlapping sets.
>   - If the plugin does not respond with a `labels` command in phase 2, clients MUST treat the plugin's recipient stanzas has having an empty label set.

That means if the client does not advertise support for the labels extension, the plugin knows that if it proceeds apace, it will be forced to use the empty label set, which would be in violation of its desired non-empty label set. _Maybe_ some plugins would accept that kind of fallback? But because plugins only see their relevant recipients, they cannot know what other recipients they will be combined with (the precise reason we designed the labels extension).

If we think this should be a per-plugin decision, I could add an associated constant to the trait (or alter the new `labels` method to return a boolean) that forces the plugin to declare whether it would allow its labels to be ignored by older clients. But that feels a bit weird.